tmc-langs-rust/src/main/java/fi/helsinki/cs/tmc/langs/rust/RustPointsParser.java

```diff
@@ -0,0 +1,59 @@
+package fi.helsinki.cs.tmc.langs.rust;
+
+import fi.helsinki.cs.tmc.langs.domain.ExerciseDesc;
+import fi.helsinki.cs.tmc.langs.domain.TestDesc;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Scanner;
+
+public class RustPointsParser {
+
+    /**
+     * Parses a file containing information of points of exercise.
+     */
+    public Optional<ExerciseDesc> parse(Scanner scanner, String exerciseName) {
+        Map<String, List<String>> map = new HashMap<>();
+        while (scanner.hasNextLine()) {
```

Comment:
Consider replacing with [`java.nio.file.Files#readAllLines(Path path, Charset cs)`](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#readAllLines(java.nio.file.Path,%20java.nio.charset.Charset)), unless we except the file to be large enough to cause memory issues.